contract/r/gnoswap/gns/gns.gno

```diff
@@ -0,0 +1,334 @@
+package gns
+
+import (
+	"std"
+	"strconv"
+	"strings"
+
+	"gno.land/p/demo/grc/grc20"
+	"gno.land/p/demo/ownable"
+	"gno.land/p/demo/ufmt"
+	pusers "gno.land/p/demo/users"
+
+	"gno.land/r/demo/grc20reg"
+	"gno.land/r/demo/users"
+
+	"gno.land/r/gnoswap/v1/common"
+	"gno.land/r/gnoswap/v1/consts"
+)
+
+const (
+	MAXIMUM_SUPPLY      = uint64(1_000_000_000_000_000)
+	INITIAL_MINT_AMOUNT = uint64(100_000_000_000_000)
+	MAX_EMISSION_AMOUNT = uint64(900_000_000_000_000) // MAXIMUM_SUPPLY - INITIAL_MINT_AMOUNT
+)
+
+var (
+	owner         *ownable.Ownable
+	Token         *grc20.Token
+	privateLedger *grc20.PrivateLedger
+	UserTeller    grc20.Teller
+
+	leftEmissionAmount   uint64
+	mintedEmissionAmount uint64
+	lastMintedHeight     int64
+
+	burnAmount uint64
+)
+
+func init() {
+	owner = ownable.NewWithAddress(consts.ADMIN)
+	Token, privateLedger = grc20.NewToken("Gnoswap", "GNS", 6)
+	UserTeller = Token.CallerTeller()
+
+	privateLedger.Mint(owner.Owner(), INITIAL_MINT_AMOUNT)
+	getter := func() *grc20.Token { return Token }
+	grc20reg.Register(getter, "")
+
+	// Initial amount set to 900_000_000_000_000 (MAXIMUM_SUPPLY - INITIAL_MINT_AMOUNT).
+	// leftEmissionAmount will decrease as tokens are minted.
+	setLeftEmissionAmount(MAX_EMISSION_AMOUNT)
+	setMintedEmissionAmount(uint64(0))
+	setLastMintedHeight(std.GetHeight())
+	burnAmount = uint64(0)
+}
+
+func GetName() string {
+	return Token.GetName()
+}
+
+func GetSymbol() string {
+	return Token.GetSymbol()
+}
+
+func GetDecimals() uint {
+	return Token.GetDecimals()
+}
+
+func TotalSupply() uint64 {
+	return Token.TotalSupply()
+}
+
+func KnownAccounts() int {
+	return Token.KnownAccounts()
+}
+
+func BalanceOfAddress(owner std.Address) uint64 {
+	common.AssertValidAddr(owner)
+	return Token.BalanceOf(owner)
+}
+
+func AllowanceOfAddress(owner, spender std.Address) uint64 {
+	common.AssertValidAddr(owner)
+	common.AssertValidAddr(spender)
+	return Token.Allowance(owner, spender)
+}
```

Comment:
It seems you should just expose the UserTeller, which has, if I'm not mistaken, the same checks.

Reply:
removing pr
https://github.com/gnoswap-labs/gnoswap/pull/482